wilsonChan/assignments/cyclicSort/lc442/lc442.py

```diff
@@ -0,0 +1,101 @@
+# Problem Statement #
+
+# We are given an unsorted array containing ‘n’ numbers taken from the range 1 to ‘n’. 
+# The array has some numbers appearing twice, find all these duplicate numbers without using any extra space.
+
+
+'''
+Misc. Notes
+-"without using extra space" (space constraint)          space O(n)
+-range 1 to ‘n’       (focus on actual value inside array, not index)
+
+
+OUTPUT:
+[3, 4, 4, 5, 5]                 #[4,5]
+[5, 4, 7, 2, 3, 5, 3]           #[3]
+
+[3, 4, 4, , 70, 5, 5,100]       # not possible
+
+
+Variables
+-i = 0       (starting value and its value will change throughout)
+    -The approach doesn't necessarily require initialization
+
+
+Steps:
+0) Manipulate the given input values
+1) Iterate through the array with for loop
+    -No need to initialize variable for this approach
+    -In this approach, sorting the values isn't necessary since the problem
+     is asking to return the duplicate values
+2) The goal is to make the values negative
+    -Easier to identify that there is a duplicate
+      -If multiple negative values are discovered, it indicates that the
+       value has already been seen
+          -The duplicate values will share the same index reference and will
+           be appended into duplicateNumbers empty array
+
+
+
+QUESTION/ASSUMPTION:
+-Will there be an empty array?
+    -if statement to check if there is anything within the array
+      -return []
```

Comment:
I love that you wrote out edge cases, and your steps for your approach. However, this approach does not mention that you will be creating an array to store these negative values. A bit more detail in the approach would have shown that it is using extra space, and therefore not meeting the prompt's requirements.